docs/design/value-attribution.md

```diff
@@ -0,0 +1,79 @@
+# Value Attribution
+
+### Brandon Iglesias, Dennis Coyle
+
+
+## Definitions
+
+*Attribution* - _Acknowledgment for establishing the relationship between Storj and a customer_
+
+## OVERVIEW
+
+When our partners bring data to the Tardigrade network we want the ability to attach attribution for the data. 
+
+Our partners will have connectors that their customers will use to store data on the Tardigrade network. When one of their customers uses the connector to store data in a bucket on the Tardigrade network, we want to give that partner attribution for the data that is stored in it.
+
+## GOALS
+
+
+* A generic library that can be used in all  Golang third party connectors
+
+* Data attribution on a per bucket basis
+
+* Ability to calculate how much data was stored in the bucket for a given amount of time. 
+
+* Ability to calculate attributed data bandwidth on the Storj Network
+
+
+## NON GOALS
+
+* Attributions per object
+
+* Multiple attributions per bucket
+
+* Attributions per project
+
+* Attributions per satellite account
+
+
+
+## SCENARIOS
+
+
+* A client points partner connector A to a bucket with NO data in it.
+
+	* The partner receives the attribution for all of the data that the client stores and retrieves from it. 
+
+	* The bucket must be empty when the partner connector is connected to it. 
+
+* A client points partner connector A to an existing bucket that NO OTHER connector has pointed to but has data in it. 
+
+	* The partner does NOT receive attribution for data stored or retrieved from that bucket. 
+
+* A client points partner connector A to an existing bucket that partner connector B had already been pointed to. 
+
+	* Partner connector B receives the attribution for all of the data that is stored and retrieved from the bucket. 
+
+## DESIGN
+### Connector
+Each partner will have a registered id, (which we will refer to as the partner id) that will identify a partners connector on the Storj network.  When a user uploads data to a specified bucket through the connector,  the connector will include the partner id in the context of the request. Before an upload occurs, the uplink will communicate the partner id and bucket name with the tardigrade satellite, checking for a previous attribution. If no attribution is found on the specified bucket and the bucket is currently void of data, the satellite will attribute the partners id to that bucket within the metadata struct. Concurrently to updating the metadata struct the satelitte will add the necessary data to the Attribution table. 
+
+### Database
+The attribution table will consist of data that allows for ease of calculating total attribution for a partner. 
+
+| Name  | Type |
+| ------------- | ------------- |
+| bucket_id (pk) | uuid  | 
+| user_id  | uuid  |
+| partner_id  | uuid  |
```

Comment:
maybe this is a string instead of a uuid?

Reply:
My issue with strings as IDs is historically I've run into normalization issues. If you think the space saving is a better tradeoff than clarity I'm all for swapping

Reply:
if the ID is indeed a user-agent though, it's not a UUID

Reply:
or is there another table of partner id to partner info mappings?

Reply:
My thought was we would have a map of `[valid user-agent]uuid` since we authorize partners before allowing them attribution, we would add them to that list.

Reply:
and not a table (although we could) but a config on satellite

Reply:
benefit of this would be if a partner has multiple user-agents

Reply:
My knowledge of the system is lacking still so this may be a dumb question but do we have to worry about someone say taking `uplink` and modifying it to generate millions of user-agents that would then cause our attribution service to crank on endless amounts of fake work and causing a lot of load on our database?

Reply:
They would have to have a signed certificate to upload anything so this is prevented same as we would do for any user trying to ddos us. This case in particular though, the mappings of user agents to partner ids would bypass any value attribution if the user-agent was not an authorized partner.